File: app/src/main/java/com/alamodrafthouse/injection/component/ActivityComponent.java
package com.alamodrafthouse.injection.component;

import com.alamodrafthouse.injection.module.ActivityModule;
import com.alamodrafthouse.injection.scope.ActivityScope;
import com.alamodrafthouse.ui.category.CategoryActivity;
import com.alamodrafthouse.ui.kotlinsample.KotlinActivity;
import dagger.Component;

/**
 * Created by Shekar on 3/3/17.
 */

@ActivityScope @Component(dependencies = ApplicationComponent.class, modules = {
    ActivityModule.class
}) public interface ActivityComponent {
  void inject(CategoryActivity categoryActivity);

  void inject(KotlinActivity kotlinActivity);
}

File: app/src/main/java/com/mutualmobile/praxis/injection/component/ActivityComponent.kt
package com.mutualmobile.praxis.injection.component

import com.mutualmobile.praxis.injection.module.ActivityModule
import com.mutualmobile.praxis.injection.module.FragmentModule
import com.mutualmobile.praxis.injection.scope.ActivityScope
import com.mutualmobile.praxis.ui.kotlinsample.KotlinActivity
import dagger.Subcomponent

@ActivityScope @Subcomponent(modules = arrayOf(ActivityModule::class)) interface ActivityComponent {

  fun plusFragmentComponent(fragmentModule: FragmentModule): FragmentComponent

  fun inject(kotlinActivity: KotlinActivity)
}
